File: LabTab-Back/src/main/java/cl/labtab/api/dtos/request/CreateBillRequest.java
package cl.labtab.api.dtos.request;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;
import java.util.UUID;

public record CreateBillRequest(
        @NotNull UUID dineSessionId,
        @PositiveOrZero BigDecimal serviceChargePct
) {
}

File: labtab_app_v1/lib/features/bill/data/models/create_bill_request.dart
class CreateBillRequest {
  final String dineSessionId;
  final double serviceChargePct;

  const CreateBillRequest({
    required this.dineSessionId,
    this.serviceChargePct = 0.0,
  });

  Map<String, dynamic> toJson() => {
        'dineSessionId': dineSessionId,
        'serviceChargePct': serviceChargePct,
      };
}
